### PR TITLE
Visprep

### PR DESCRIPTION
**Pull request by blsemo**
Add prep steps for VGA to command line

### CODE REVIEW

File: cliTest/testvisprepparser.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        }
+        ArgumentHolder ah{"prog", "-pg", "0.1", "-pp", "0.1"};
+        REQUIRE_THROWS_WITH(parser.parse(ah.argc(), ah.argv()), Catch::Contains("Error parsing line: 0.1"));
+    }
+
+    SECTION("Nonsensical visiblity restriction")
+    {
+        VisPrepParser parser;
+        ArgumentHolder ah{"prog", "-pr", "foo"};
+        REQUIRE_THROWS_WITH(parser.parse(ah.argc(), ah.argv()), Catch::Contains("Restricted visibilyt of 'foo' makes no sense, use a positive number or -1 for unrestricted"));
```
Comment (pklampros) [inline cliTest/testvisprepparser.cpp]: Minor: visibilyt -> visibility

File: cliTest/testvisprepparser.cpp [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    }
+
+    SECTION("Nonsensical visiblity restriction")
+    {
+        VisPrepParser parser;
+        ArgumentHolder ah{"prog", "-pr", "0.0"};
+        REQUIRE_THROWS_WITH(parser.parse(ah.argc(), ah.argv()), Catch::Contains("Restricted visibilyt of '0.0' makes no sense, use a positive number or -1 for unrestricted"));
```
Comment (pklampros) [inline cliTest/testvisprepparser.cpp]: Minor: visibilyt -> visibility

File: depthmapXcli/visprepparser.cpp [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            pointFile = argv[i];
+        }
+        else if ( strcmp("-pr", argv[i]) == 0)
+        {
+            ENFORCE_ARGUMENT("-pr", i);
+            m_maxVisibility = std::atof(argv[i]);
+            if ( m_maxVisibility == 0.0)
+            {
+                std::stringstream message;
+                message << "Restricted visibilyt of '" << argv[i] << "' makes no sense, use a positive number or -1 for unrestricted";
```
Comment (pklampros) [inline depthmapXcli/visprepparser.cpp]: Minor: visibilyt -> visibility

File: depthmapXcli/runmethods.cpp [hunk 4]
```diff
@@ -105,4 +106,60 @@ namespace dm_runmethods
...
+            throw depthmapX::RuntimeException("Graph must have line data before preparing VGA");
+        }
+        // set grid
+        QtRegion r = mGraph->getRegion();
+
+        GridProperties gp(__max(r.width(), r.height()));
+        if ( gridSize > gp.getMax() ||  gridSize < gp.getMin())
+        {
+            std::stringstream message;
+            message << "Chosen grid spacing " << gridSize << " is outside of the expected intervall of "
```
Comment (pklampros) [inline depthmapXcli/runmethods.cpp]: Minor: intervall -> interval

Comment (pklampros) [reply depthmapXcli/runmethods.cpp]: Also, some tests for this exception?

**Comment by blsemo** [comment]
Tricky, as run methods aren't really testable. That's why I tried out the calculation to a separate class in sala.

On May 19, 2017 4:22:05 PM GMT+01:00, Petros Koutsolampros <notifications@github.com> wrote:
>orange-vertex commented on this pull request.
>
>
>
>> +        auto mGraph =
>loadGraph(clp.getFileName().c_str(),perfWriter);
>+
>+        std::cout << "Initial checks... " << std::flush;
>+        auto state = mGraph->getState();
>+        if (~state & MetaGraph::LINEDATA)
>+        {
>+            throw depthmapX::RuntimeException("Graph must have line
>data before preparing VGA");
>+        }
>+        // set grid
>+        QtRegion r = mGraph->getRegion();
>+
>+        GridProperties gp(__max(r.width(), r.height()));
>+        if ( gridSize > gp.getMax() ||  gridSize < gp.getMin())
>+        {
>+            std::stringstream message;
>+            message << "Chosen grid spacing " << gridSize << " is
>outside of the expected intervall of "
>
>Also, some tests for this exception?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/blsemo/depthmapX/pull/21#discussion_r117507065

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Pull request merged**